tests/unit/python/foglamp/services/core/api/test_notification.py

```diff
@@ -0,0 +1,715 @@
+        return json.dumps({"result": "OK"})
+
+
+@asyncio.coroutine
+def mock_read_category_val(key):
+    if key.endswith("ruleTest Notification"):
+        return rule_config[notification_config['rule']['value']]
+    if key.endswith("deliveryTest Notification"):
+        return delivery_config[notification_config['channel']['value']]
+    if key.endswith("Test Notification"):
+        return notification_config
+    return ""
+
+
+@asyncio.coroutine
+def mock_read_all_child_category_names():
+    return [{
+        "parent": "Notifications",
+        "child": "Test Notification",
+    }]
+
+
+@asyncio.coroutine
+def mock_create_category():
+    return ""
+
+
+@asyncio.coroutine
+def mock_update_category():
+    return ""
+
+
+@asyncio.coroutine
+def mock_create_child_category():
+    return ""
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("core", "api", "notification")
+class TestNotification:
+    @pytest.fixture
+    def client(self, loop, test_client):
+        app = web.Application(loop=loop)
+        routes.setup(app)
+        return loop.run_until_complete(test_client(app))
+
+    async def test_get_plugin(self, mocker, client):
+        rules_and_delivery = {'rules': rule_config, 'delivery': delivery_config}
+        mocker.patch.object(ServiceRegistry, 'get', return_value=mock_registry)
+        mocker.patch.object(notification, '_hit_get_url', side_effect=[mock_get_url("/foglamp/notification/rules"),
+                                                                       mock_get_url("/foglamp/notification/delivery")])
+
+        resp = await client.get('/foglamp/notification/plugin')
+        assert 200 == resp.status
+        result = await resp.text()
+        json_response = json.loads(result)
+        assert rules_and_delivery == json_response
+
+    async def test_get_notification(self, mocker, client):
+        notif = {
+            "name": notification_config['name']['value'],
+            "description": notification_config['description']['value'],
+            "rule": notification_config['rule']['value'],
+            "ruleConfig": rule_config[notification_config['rule']['value']],
+            "channel": notification_config['channel']['value'],
+            "deliveryConfig": delivery_config[notification_config['channel']['value']],
+            "notificationType": notification_config['notification_type']['value'],
+            "enable": notification_config['enable']['value'],
+        }
+
+        mocker.patch.object(connect, 'get_storage_async')
+        mocker.patch.object(ConfigurationManager, '__init__', return_value=None)
+        mocker.patch.object(ConfigurationManager, '_read_category_val', side_effect=[
+            mock_read_category_val("Test Notification"),
+            mock_read_category_val("ruleTest Notification"),
+            mock_read_category_val("deliveryTest Notification")])
+
+        resp = await client.get('/foglamp/notification/Test Notification')
```

Comment:
@amarendra-dianomic this should be `await client.get('/foglamp/notification/Test%20Notification')`